### DISCUSSION TITLE
How to avoid quarto positioning figures to the end of the chapter when rendering a pdf book?

### DISCUSSION THREAD

**Discussion by barizona** [quarto-dev/quarto-cli#2182] [Q&A] [answered]
Hi,
Do you know how to avoid quarto positioning figures to the end of the chapter when rendering a pdf book? 
I inserted the figure like this:
`![Long caption .... ](fig_name.png){#fig-name width="65%" fig.pos="h"}`

When I decrease width to 60% it appears in its correct place. But if I increase quarto put it to the end of the chapter.

This is my YAML file:
`project:
  type: book

book:
  title: "Title"
  author: "My Name"
  date: 2022-08-30
  date-format: "YYYY"

  chapters:
    - index.qmd
    - chapter1.qmd
    - chapter2.qmd
    - chapter3.qmd
    
bibliography: references.bib

execute:
  echo: false

lang: hu
language: 
  crossref-fig-title: "ábra"
  crossref-tbl-title: "táblázat"

format:

  pdf:
    documentclass: scrreprt
    pdf-engine: xelatex
    papersize: A4
    pandoc_args: --listings
    classoption: notitlepage
    keep-tex: true
    toc: true
    toc-title: "Tartalomjegyzék"
    toc-depth: 3
    number-sections: false
    colorlinks: true
    mainfont: Calluna
    sansfont: Calluna
    urlcolor: DodgerBlue4
    linkcolor: DodgerBlue4
    citecolor: DodgerBlue4
    fig-pos: "h"
    fig-cap-location: bottom
    linestretch: 1.2

editor: visual

header-includes:
  - \usepackage{pdfpages}
  - \usepackage{microtype}
  - \setlength{\parindent}{6mm}
  - \pretolerance=10000
  - \usepackage{colortbl}`
Thanks

**Comment by rgaiacs** [comment]
>  Do you know how to avoid quarto positioning figures to the end of the chapter when rendering a pdf book?

LaTeX will calculate the best position for the figure. You can tell LaTeX what is your preference using `fig.pos` as you did but it does **not** guarantee that LaTeX will respect it because it's possible that the page does not have enough space for the figure.

**Reply by barizona** [reply]
But the figure can fit to a single page at the end of the chapter? (Yes it can...)

**Comment by mcanouil** [comment, marked as answer]
You might want to use `fig.pos = "h!"` or `fig.pos = "H"`.

**Reply by barizona** [reply]
Thanks for the suggestion. I've tried, the result is the same (fig at the end of the chapter).

**Reply by barizona** [reply]
But when I've corrected this part in the YAML: 
`fig-pos: "h!"` it worked! Thanks for the tip.

**Reply by mcanouil** [reply]
By the way, this is a LaTeX behaviour, not a Quarto issue/behaviour.

**Reply by barizona** [reply]
Yes, I understood it, finally.
